<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="20" total="66" passed="37" failed="9" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2025-04-14T02:39:52 GMT+02:00" name="E-commerce Suite" finished-at="2025-04-14T02:41:14 GMT+02:00" duration-ms="82064">
    <groups>
    </groups>
    <test started-at="2025-04-14T02:39:52 GMT+02:00" name="FirstTests" finished-at="2025-04-14T02:40:41 GMT+02:00" duration-ms="48786">
      <class name="tests.LoginTest">
        <test-method signature="testInValidLogin(java.lang.String,java.lang.String,java.lang.String)[pri:1, instance:tests.LoginTest@5762806e]" started-at="2025-04-14T02:39:52 GMT+02:00" name="testInValidLogin" data-provider="invalidData" finished-at="2025-04-14T02:39:58 GMT+02:00" duration-ms="5740" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA[]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[ðŸ˜ž: Username is required]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Error message does not match for locked out user expected [ðŸ˜ž: Username is required] but found [Epic sadface: Username is required]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Error message does not match for locked out user expected [ðŸ˜ž: Username is required] but found [Epic sadface: Username is required]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:656)
at pages.LoginPage.login(LoginPage.java:42)
at tests.LoginTest.testInValidLogin(LoginTest.java:46)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at tests.TestRunner.main(TestRunner.java:25)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testInValidLogin -->
        <test-method signature="testInValidLogin(java.lang.String,java.lang.String,java.lang.String)[pri:1, instance:tests.LoginTest@5762806e]" started-at="2025-04-14T02:39:58 GMT+02:00" name="testInValidLogin" data-provider="invalidData" finished-at="2025-04-14T02:39:59 GMT+02:00" duration-ms="856" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA[invalid_user]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[ðŸ˜ž: Password is required]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Error message does not match for locked out user expected [ðŸ˜ž: Password is required] but found [Epic sadface: Password is required]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Error message does not match for locked out user expected [ðŸ˜ž: Password is required] but found [Epic sadface: Password is required]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:656)
at pages.LoginPage.login(LoginPage.java:42)
at tests.LoginTest.testInValidLogin(LoginTest.java:46)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at tests.TestRunner.main(TestRunner.java:25)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testInValidLogin -->
        <test-method signature="testInValidLogin(java.lang.String,java.lang.String,java.lang.String)[pri:1, instance:tests.LoginTest@5762806e]" started-at="2025-04-14T02:39:59 GMT+02:00" name="testInValidLogin" data-provider="invalidData" finished-at="2025-04-14T02:40:00 GMT+02:00" duration-ms="818" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA[standard_user]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[wrong_pass]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[ðŸ˜ž: Username and password do not match any user in this service]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Error message does not match for locked out user expected [ðŸ˜ž: Username and password do not match any user in this service] but found [Epic sadface: Username and password do not match any user in this service]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Error message does not match for locked out user expected [ðŸ˜ž: Username and password do not match any user in this service] but found [Epic sadface: Username and password do not match any user in this service]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:656)
at pages.LoginPage.login(LoginPage.java:42)
at tests.LoginTest.testInValidLogin(LoginTest.java:46)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at tests.TestRunner.main(TestRunner.java:25)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testInValidLogin -->
        <test-method signature="testInValidLogin(java.lang.String,java.lang.String,java.lang.String)[pri:1, instance:tests.LoginTest@5762806e]" started-at="2025-04-14T02:40:00 GMT+02:00" name="testInValidLogin" data-provider="invalidData" finished-at="2025-04-14T02:40:00 GMT+02:00" duration-ms="802" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA[test]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[1234]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[ðŸ˜ž: Username and password do not match any user in this service]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Error message does not match for locked out user expected [ðŸ˜ž: Username and password do not match any user in this service] but found [Epic sadface: Username and password do not match any user in this service]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Error message does not match for locked out user expected [ðŸ˜ž: Username and password do not match any user in this service] but found [Epic sadface: Username and password do not match any user in this service]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:656)
at pages.LoginPage.login(LoginPage.java:42)
at tests.LoginTest.testInValidLogin(LoginTest.java:46)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at tests.TestRunner.main(TestRunner.java:25)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testInValidLogin -->
        <test-method signature="testValidLogin(java.lang.String,java.lang.String,java.lang.String)[pri:2, instance:tests.LoginTest@5762806e]" started-at="2025-04-14T02:40:00 GMT+02:00" name="testValidLogin" data-provider="validData" finished-at="2025-04-14T02:40:01 GMT+02:00" duration-ms="994" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[standard_user]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[secret_sauce]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[https://www.saucedemo.com/inventory.html]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testValidLogin -->
      </class> <!-- tests.LoginTest -->
      <class name="tests.home.Navigation">
        <test-method signature="navigateProducts()[pri:0, instance:tests.home.Navigation@50a7bc6e]" started-at="2025-04-14T02:40:01 GMT+02:00" name="navigateProducts" finished-at="2025-04-14T02:40:16 GMT+02:00" duration-ms="14039" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- navigateProducts -->
      </class> <!-- tests.home.Navigation -->
      <class name="tests.home.MenuTest">
        <test-method signature="testOpenMenu()[pri:1, instance:tests.home.MenuTest@4034c28c]" started-at="2025-04-14T02:40:34 GMT+02:00" name="testOpenMenu" finished-at="2025-04-14T02:40:34 GMT+02:00" duration-ms="351" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testOpenMenu -->
        <test-method signature="testSideBarLinks(java.lang.String,java.lang.String)[pri:2, instance:tests.home.MenuTest@4034c28c]" started-at="2025-04-14T02:40:34 GMT+02:00" name="testSideBarLinks" data-provider="sidebar" finished-at="2025-04-14T02:40:34 GMT+02:00" duration-ms="304" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[inventory_sidebar_link]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[https://www.saucedemo.com/inventory.html]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testSideBarLinks -->
        <test-method signature="testSideBarLinks(java.lang.String,java.lang.String)[pri:2, instance:tests.home.MenuTest@4034c28c]" started-at="2025-04-14T02:40:34 GMT+02:00" name="testSideBarLinks" data-provider="sidebar" finished-at="2025-04-14T02:40:40 GMT+02:00" duration-ms="5711" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[about_sidebar_link]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[https://saucelabs.com/]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testSideBarLinks -->
        <test-method signature="testSideBarLinks(java.lang.String,java.lang.String)[pri:2, instance:tests.home.MenuTest@4034c28c]" started-at="2025-04-14T02:40:40 GMT+02:00" name="testSideBarLinks" data-provider="sidebar" finished-at="2025-04-14T02:40:41 GMT+02:00" duration-ms="385" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA[reset_sidebar_link]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[https://www.saucedemo.com/inventory.html]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA['Remove' buttons should not be visible after removing all items. expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: 'Remove' buttons should not be visible after removing all items. expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertTrue(Assert.java:57)
at pages.home.HomePage.sideBarLinks(HomePage.java:81)
at tests.home.MenuTest.testSideBarLinks(MenuTest.java:36)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at tests.TestRunner.main(TestRunner.java:25)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testSideBarLinks -->
        <test-method signature="testSideBarLinks(java.lang.String,java.lang.String)[pri:2, instance:tests.home.MenuTest@4034c28c]" started-at="2025-04-14T02:40:41 GMT+02:00" name="testSideBarLinks" data-provider="sidebar" finished-at="2025-04-14T02:40:41 GMT+02:00" duration-ms="296" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[logout_sidebar_link]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[https://www.saucedemo.com/]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testSideBarLinks -->
      </class> <!-- tests.home.MenuTest -->
      <class name="tests.home.FooterTest">
        <test-method signature="testFooter(java.lang.String,java.lang.String)[pri:1, instance:tests.home.FooterTest@17c1bced]" started-at="2025-04-14T02:40:25 GMT+02:00" name="testFooter" data-provider="pathSocial" finished-at="2025-04-14T02:40:25 GMT+02:00" duration-ms="507" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[//*[@id="page_wrapper"]/footer/ul/li[1]/a]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[https://twitter.com/saucelabs]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testFooter -->
        <test-method signature="testFooter(java.lang.String,java.lang.String)[pri:1, instance:tests.home.FooterTest@17c1bced]" started-at="2025-04-14T02:40:25 GMT+02:00" name="testFooter" data-provider="pathSocial" finished-at="2025-04-14T02:40:26 GMT+02:00" duration-ms="622" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[//*[@id="page_wrapper"]/footer/ul/li[2]/a]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[https://www.facebook.com/saucelabs]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testFooter -->
        <test-method signature="testFooter(java.lang.String,java.lang.String)[pri:1, instance:tests.home.FooterTest@17c1bced]" started-at="2025-04-14T02:40:26 GMT+02:00" name="testFooter" data-provider="pathSocial" finished-at="2025-04-14T02:40:26 GMT+02:00" duration-ms="586" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[//*[@id="page_wrapper"]/footer/ul/li[3]/a]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[https://www.linkedin.com/company/sauce-labs/]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testFooter -->
      </class> <!-- tests.home.FooterTest -->
      <class name="tests.home.FiltrationTest">
        <test-method signature="testNameSorting(java.lang.String,java.lang.String)[pri:0, instance:tests.home.FiltrationTest@161b062a]" started-at="2025-04-14T02:40:16 GMT+02:00" name="testNameSorting" data-provider="filterName" finished-at="2025-04-14T02:40:18 GMT+02:00" duration-ms="2279" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[az]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[Products are not sorted by name A to Z.]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testNameSorting -->
        <test-method signature="testNameSorting(java.lang.String,java.lang.String)[pri:0, instance:tests.home.FiltrationTest@161b062a]" started-at="2025-04-14T02:40:18 GMT+02:00" name="testNameSorting" data-provider="filterName" finished-at="2025-04-14T02:40:20 GMT+02:00" duration-ms="2345" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[za]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[Products are not sorted by name Z to A.]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testNameSorting -->
        <test-method signature="testPriceSorting(java.lang.String,java.lang.String)[pri:0, instance:tests.home.FiltrationTest@161b062a]" started-at="2025-04-14T02:40:20 GMT+02:00" name="testPriceSorting" data-provider="filterPrice" finished-at="2025-04-14T02:40:22 GMT+02:00" duration-ms="2293" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[lohi]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[Prices are not sorted from low to high.]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testPriceSorting -->
        <test-method signature="testPriceSorting(java.lang.String,java.lang.String)[pri:0, instance:tests.home.FiltrationTest@161b062a]" started-at="2025-04-14T02:40:22 GMT+02:00" name="testPriceSorting" data-provider="filterPrice" finished-at="2025-04-14T02:40:25 GMT+02:00" duration-ms="2297" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[hilo]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[Prices are not sorted from high to low.]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testPriceSorting -->
      </class> <!-- tests.home.FiltrationTest -->
      <class name="tests.home.AddToCart">
        <test-method signature="addProductsToCart()[pri:0, instance:tests.home.AddToCart@2d9d4f9d]" started-at="2025-04-14T02:40:26 GMT+02:00" name="addProductsToCart" finished-at="2025-04-14T02:40:34 GMT+02:00" duration-ms="7380" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- addProductsToCart -->
      </class> <!-- tests.home.AddToCart -->
    </test> <!-- FirstTests -->
    <test started-at="2025-04-14T02:40:41 GMT+02:00" name="SecondTests" finished-at="2025-04-14T02:41:04 GMT+02:00" duration-ms="22800">
      <class name="tests.YourCartTest">
        <test-method signature="testGoToYourCart()[pri:1, instance:tests.YourCartTest@147ed70f]" started-at="2025-04-14T02:40:50 GMT+02:00" name="testGoToYourCart" finished-at="2025-04-14T02:40:50 GMT+02:00" duration-ms="205" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testGoToYourCart -->
        <test-method signature="testClickOnContinueShopping()[pri:2, instance:tests.YourCartTest@147ed70f]" started-at="2025-04-14T02:40:50 GMT+02:00" name="testClickOnContinueShopping" finished-at="2025-04-14T02:40:51 GMT+02:00" duration-ms="257" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testClickOnContinueShopping -->
        <test-method signature="testTitleOfAllProductInCart()[pri:3, instance:tests.YourCartTest@147ed70f]" started-at="2025-04-14T02:40:51 GMT+02:00" name="testTitleOfAllProductInCart" finished-at="2025-04-14T02:40:54 GMT+02:00" duration-ms="2887" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testTitleOfAllProductInCart -->
        <test-method signature="testClickOnCheckout()[pri:5, instance:tests.YourCartTest@147ed70f]" started-at="2025-04-14T02:40:54 GMT+02:00" name="testClickOnCheckout" finished-at="2025-04-14T02:40:54 GMT+02:00" duration-ms="195" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testClickOnCheckout -->
      </class> <!-- tests.YourCartTest -->
      <class name="tests.LoginTest">
        <test-method signature="testValidLogin(java.lang.String,java.lang.String,java.lang.String)[pri:2, instance:tests.LoginTest@29176cc1]" started-at="2025-04-14T02:40:41 GMT+02:00" name="testValidLogin" data-provider="validData" finished-at="2025-04-14T02:40:42 GMT+02:00" duration-ms="833" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[standard_user]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[secret_sauce]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[https://www.saucedemo.com/inventory.html]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testValidLogin -->
      </class> <!-- tests.LoginTest -->
      <class name="tests.checkout.CheckOutOverViewTest">
        <test-method signature="testClickOnCancelBTN()[pri:1, instance:tests.checkout.CheckOutOverViewTest@124c278f]" started-at="2025-04-14T02:41:00 GMT+02:00" name="testClickOnCancelBTN" finished-at="2025-04-14T02:41:00 GMT+02:00" duration-ms="355" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testClickOnCancelBTN -->
        <test-method signature="testCartCalculations()[pri:2, instance:tests.checkout.CheckOutOverViewTest@124c278f]" started-at="2025-04-14T02:41:00 GMT+02:00" name="testCartCalculations" finished-at="2025-04-14T02:41:01 GMT+02:00" duration-ms="401" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCartCalculations -->
        <test-method signature="testTitleOfAllProductInCart()[pri:3, instance:tests.checkout.CheckOutOverViewTest@124c278f]" started-at="2025-04-14T02:41:01 GMT+02:00" name="testTitleOfAllProductInCart" finished-at="2025-04-14T02:41:04 GMT+02:00" duration-ms="2712" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testTitleOfAllProductInCart -->
        <test-method signature="testClickOnFinishBTN()[pri:4, instance:tests.checkout.CheckOutOverViewTest@124c278f]" started-at="2025-04-14T02:41:04 GMT+02:00" name="testClickOnFinishBTN" finished-at="2025-04-14T02:41:04 GMT+02:00" duration-ms="354" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testClickOnFinishBTN -->
      </class> <!-- tests.checkout.CheckOutOverViewTest -->
      <class name="tests.checkout.CheckOutInfoTest">
        <test-method signature="testGoBack()[pri:2, instance:tests.checkout.CheckOutInfoTest@61dd025]" started-at="2025-04-14T02:40:54 GMT+02:00" name="testGoBack" finished-at="2025-04-14T02:40:54 GMT+02:00" duration-ms="239" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testGoBack -->
        <test-method signature="testInValidFormSubmission(java.lang.String,java.lang.String,java.lang.String,java.lang.String)[pri:3, instance:tests.checkout.CheckOutInfoTest@61dd025]" started-at="2025-04-14T02:40:54 GMT+02:00" name="testInValidFormSubmission" data-provider="invalidData" finished-at="2025-04-14T02:40:55 GMT+02:00" duration-ms="1227" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA[]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[Last]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[12345]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Error: First Name is required]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Last Name error icon should NOT be visible expected [false] but found [true]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Last Name error icon should NOT be visible expected [false] but found [true]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertFalse(Assert.java:79)
at pages.checkout.CheckOutInfoPage.inValidFormSubmission(CheckOutInfoPage.java:82)
at tests.checkout.CheckOutInfoTest.testInValidFormSubmission(CheckOutInfoTest.java:49)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at tests.TestRunner.main(TestRunner.java:25)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testInValidFormSubmission -->
        <test-method signature="testInValidFormSubmission(java.lang.String,java.lang.String,java.lang.String,java.lang.String)[pri:3, instance:tests.checkout.CheckOutInfoTest@61dd025]" started-at="2025-04-14T02:40:55 GMT+02:00" name="testInValidFormSubmission" data-provider="invalidData" finished-at="2025-04-14T02:40:57 GMT+02:00" duration-ms="1288" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA[First]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[12345]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Error: Last Name is required]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[First Name error icon should NOT be visible expected [false] but found [true]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: First Name error icon should NOT be visible expected [false] but found [true]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertFalse(Assert.java:79)
at pages.checkout.CheckOutInfoPage.inValidFormSubmission(CheckOutInfoPage.java:76)
at tests.checkout.CheckOutInfoTest.testInValidFormSubmission(CheckOutInfoTest.java:49)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at tests.TestRunner.main(TestRunner.java:25)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testInValidFormSubmission -->
        <test-method signature="testInValidFormSubmission(java.lang.String,java.lang.String,java.lang.String,java.lang.String)[pri:3, instance:tests.checkout.CheckOutInfoTest@61dd025]" started-at="2025-04-14T02:40:57 GMT+02:00" name="testInValidFormSubmission" data-provider="invalidData" finished-at="2025-04-14T02:40:58 GMT+02:00" duration-ms="1241" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA[First]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[Last]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Error: Postal Code is required]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[First Name error icon should NOT be visible expected [false] but found [true]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: First Name error icon should NOT be visible expected [false] but found [true]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertFalse(Assert.java:79)
at pages.checkout.CheckOutInfoPage.inValidFormSubmission(CheckOutInfoPage.java:76)
at tests.checkout.CheckOutInfoTest.testInValidFormSubmission(CheckOutInfoTest.java:49)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at tests.TestRunner.main(TestRunner.java:25)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testInValidFormSubmission -->
        <test-method signature="testInValidFormSubmission(java.lang.String,java.lang.String,java.lang.String,java.lang.String)[pri:3, instance:tests.checkout.CheckOutInfoTest@61dd025]" started-at="2025-04-14T02:40:58 GMT+02:00" name="testInValidFormSubmission" data-provider="invalidData" finished-at="2025-04-14T02:40:59 GMT+02:00" duration-ms="1094" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Error: First Name is required]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testInValidFormSubmission -->
        <test-method signature="testValidFormSubmission(java.lang.String,java.lang.String,java.lang.String,java.lang.String)[pri:4, instance:tests.checkout.CheckOutInfoTest@61dd025]" started-at="2025-04-14T02:40:59 GMT+02:00" name="testValidFormSubmission" data-provider="validData" finished-at="2025-04-14T02:41:00 GMT+02:00" duration-ms="1153" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[test first]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[test last]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[12345]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[https://www.saucedemo.com/checkout-step-two.html]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testValidFormSubmission -->
      </class> <!-- tests.checkout.CheckOutInfoTest -->
      <class name="tests.home.MenuTest">
        <test-method signature="testOpenMenu()[pri:1, instance:tests.home.MenuTest@2f177a4b]" started-at="2025-04-14T02:40:42 GMT+02:00" name="testOpenMenu" finished-at="2025-04-14T02:40:42 GMT+02:00" duration-ms="318" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testOpenMenu -->
        <test-method signature="testCloseMenu()[pri:3, instance:tests.home.MenuTest@2f177a4b]" started-at="2025-04-14T02:40:42 GMT+02:00" name="testCloseMenu" finished-at="2025-04-14T02:40:43 GMT+02:00" duration-ms="886" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCloseMenu -->
      </class> <!-- tests.home.MenuTest -->
      <class name="tests.home.AddToCart">
        <test-method signature="addProductsToCart()[pri:0, instance:tests.home.AddToCart@4278a03f]" started-at="2025-04-14T02:40:43 GMT+02:00" name="addProductsToCart" finished-at="2025-04-14T02:40:50 GMT+02:00" duration-ms="7077" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- addProductsToCart -->
        <test-method signature="VerifyCartCount()[pri:1, instance:tests.home.AddToCart@4278a03f]" started-at="2025-04-14T02:40:50 GMT+02:00" name="VerifyCartCount" finished-at="2025-04-14T02:40:50 GMT+02:00" duration-ms="42" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- VerifyCartCount -->
      </class> <!-- tests.home.AddToCart -->
    </test> <!-- SecondTests -->
    <test started-at="2025-04-14T02:41:04 GMT+02:00" name="ThirdTests" finished-at="2025-04-14T02:41:14 GMT+02:00" duration-ms="10117">
      <class name="tests.YourCartTest">
        <test-method signature="testGoToYourCart()[pri:1, instance:tests.YourCartTest@6ce139a4]" started-at="2025-04-14T02:41:11 GMT+02:00" name="testGoToYourCart" finished-at="2025-04-14T02:41:11 GMT+02:00" duration-ms="153" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testGoToYourCart -->
        <test-method signature="testRemoveCart()[pri:4, instance:tests.YourCartTest@6ce139a4]" started-at="2025-04-14T02:41:11 GMT+02:00" name="testRemoveCart" finished-at="2025-04-14T02:41:12 GMT+02:00" duration-ms="1196" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testRemoveCart -->
        <test-method signature="testClickOnCheckout()[pri:5, instance:tests.YourCartTest@6ce139a4]" started-at="2025-04-14T02:41:12 GMT+02:00" name="testClickOnCheckout" finished-at="2025-04-14T02:41:13 GMT+02:00" duration-ms="212" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testClickOnCheckout -->
      </class> <!-- tests.YourCartTest -->
      <class name="tests.LoginTest">
        <test-method signature="testGotTOHomePage()[pri:0, instance:tests.LoginTest@77167fb7]" started-at="2025-04-14T02:41:04 GMT+02:00" name="testGotTOHomePage" finished-at="2025-04-14T02:41:04 GMT+02:00" duration-ms="106" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testGotTOHomePage -->
      </class> <!-- tests.LoginTest -->
      <class name="tests.checkout.CheckOutOverViewTest">
        <test-method signature="testCartCalculations()[pri:2, instance:tests.checkout.CheckOutOverViewTest@2ddc8ecb]" started-at="2025-04-14T02:41:14 GMT+02:00" name="testCartCalculations" finished-at="2025-04-14T02:41:14 GMT+02:00" duration-ms="28" status="FAIL">
          <exception class="java.lang.IndexOutOfBoundsException">
            <message>
              <![CDATA[Index 0 out of bounds for length 0]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:100)
at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:106)
at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:302)
at java.base/java.util.Objects.checkIndex(Objects.java:365)
at java.base/java.util.ArrayList.get(ArrayList.java:428)
at pages.checkout.CheckOutOverViewPage.verifyCartCalculations(CheckOutOverViewPage.java:32)
at tests.checkout.CheckOutOverViewTest.testCartCalculations(CheckOutOverViewTest.java:13)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at tests.TestRunner.main(TestRunner.java:25)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.IndexOutOfBoundsException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCartCalculations -->
        <test-method signature="testClickOnFinishBTN()[pri:4, instance:tests.checkout.CheckOutOverViewTest@2ddc8ecb]" started-at="2025-04-14T02:41:14 GMT+02:00" name="testClickOnFinishBTN" finished-at="2025-04-14T02:41:14 GMT+02:00" duration-ms="311" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testClickOnFinishBTN -->
      </class> <!-- tests.checkout.CheckOutOverViewTest -->
      <class name="tests.checkout.CheckOutInfoTest">
        <test-method signature="testValidFormSubmission(java.lang.String,java.lang.String,java.lang.String,java.lang.String)[pri:4, instance:tests.checkout.CheckOutInfoTest@6973bf95]" started-at="2025-04-14T02:41:13 GMT+02:00" name="testValidFormSubmission" data-provider="validData" finished-at="2025-04-14T02:41:14 GMT+02:00" duration-ms="1107" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[test first]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[test last]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[12345]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[https://www.saucedemo.com/checkout-step-two.html]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testValidFormSubmission -->
      </class> <!-- tests.checkout.CheckOutInfoTest -->
      <class name="tests.home.AddToCart">
        <test-method signature="addProductsToCart()[pri:0, instance:tests.home.AddToCart@1fe20588]" started-at="2025-04-14T02:41:04 GMT+02:00" name="addProductsToCart" finished-at="2025-04-14T02:41:11 GMT+02:00" duration-ms="6992" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- addProductsToCart -->
      </class> <!-- tests.home.AddToCart -->
    </test> <!-- ThirdTests -->
    <test started-at="2025-04-14T02:41:14 GMT+02:00" name="CleanupTest" finished-at="2025-04-14T02:41:14 GMT+02:00" duration-ms="0">
      <class name="tests.TestCleanup">
        <test-method is-config="true" signature="afterSuite()[pri:0, instance:tests.TestCleanup@12e61fe6]" started-at="2025-04-14T02:41:14 GMT+02:00" name="afterSuite" finished-at="2025-04-14T02:41:15 GMT+02:00" duration-ms="402" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterSuite -->
      </class> <!-- tests.TestCleanup -->
    </test> <!-- CleanupTest -->
  </suite> <!-- E-commerce Suite -->
</testng-results>
